nvidia-persistenced: fails to build

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

Fails to build

### Steps to reproduce

* Set ``
* Attempt to rebuild

### Expected behaviour

Builds

### Screenshots

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/4n4bmcrhfwfnskjf2n2y0zsfh6sx71bc-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/4bacfs7zrg714ffffbjp57nsvcz6zfkq-bash-5.3p3/bin/bash CC=/nix/store/qdknxw57cwy1jkrhq7fzmiis73j42jv6-gcc-14.3.0/bin/gcc LD=/nix/store/xrwdb41dqi2ia6lr2s61w5bzfg2m71pi-binutils-2.44/bin/ld AR=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/ar NM=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/nm STRIP=/nix/store/xrwdb41dqi2ia6lr2s61w5bzfg2m71pi-binutils-2.44/bin/strip OBJCOPY=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/objcopy OBJDUMP=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/objdump READELF=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/readelf HOSTCC=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/cc HOSTCXX=/nix/store/bcw9f6r9v2fm3kv7d15fcrya0mf34xds-gcc-wrapper-14.3.0/bin/c++ HOSTAR=/nix/store/ky2x48xfid0nn5arablpwlnsngj70gws-binutils-wrapper-2.44/bin/ar HOSTLD=/nix/store/ky2x48xfid0nn5arablpwlnsngj70gws-binutils-wrapper-2.44/bin/ld ARCH=x86_64 CROSS_COMPILE= KBUILD_OUTPUT=/nix/store/gnm03zm851h35x6fwypcgw4v95l45i7i-linux-zen-6.16.1-dev/lib/modules/6.16.1-zen1/build IGNORE_PREEMPT_RT_PRESENCE=1 NV_BUILD_SUPPORTS_HMM=1 SYSSRC=/nix/store/gnm03zm851h35x6fwypcgw4v95l45i7i-linux-zen-6.16.1-dev/lib/modules/6.16.1-zen1/source SYSOUT=/nix/store/gnm03zm851h35x6fwypcgw4v95l45i7i-linux-zen-6.16.1-dev/lib/modules/6.16.1-zen1/build DATE=true
/nix/store/4bacfs7zrg714ffffbjp57nsvcz6zfkq-bash-5.3p3/bin/bash: line 1: pkg-config: command not found
   CC           command_server.c
In file included from [01m[K/nix/store/9ly5qx1lbj0q1rcsf1dc0m9n3db6ii60-glibc-2.40-66-dev/include/bits/libc-header-start.h:33[m[K,
                 from [01m[K/nix/store/9ly5qx1lbj0q1rcsf1dc0m9n3db6ii60-glibc-2.40-66-dev/include/stdlib.h:26[m[K,
                 from [01m[Kcommand_server.c:29[m[K:
[01m[K/nix/store/9ly5qx1lbj0q1rcsf1dc0m9n3db6ii60-glibc-2.40-66-dev/include/features.h:197:3:[m[K [01;35m[Kwarning: [m[K#warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wcpp-Wcpp]8;;[m[K]
  197 | # [01;35m[Kwarning[m[K "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
      |   [01;35m[K^~~~~~~[m[K
In file included from [01m[Knvidia-persistenced.h:33[m[K,
                 from [01m[Kcommand_server.c:34[m[K:
[01m[Knvpd_rpc.h:9:10:[m[K [01;31m[Kfatal error: [m[Krpc/rpc.h: No such file or directory
    9 | #include [01;31m[K<rpc/rpc.h>[m[K
      |          [01;31m[K^~~~~~~~~~~[m[K
compilation terminated.
make: *** [Makefile:144: _out/Linux_x86_64/command_server.o] Error 1
```

### Additional context

_No response_

### System metadata

- system: `"x86_64-linux"`
 - host os: `Linux 6.16.1-zen1, NixOS, 25.11 (Xantusia), 25.11.20250819.2007595`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/racci/.config/nix/nix.conf:/nix/store/r8jq1rginj1qc4yrpjgl8a3r0b254n49-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/racci/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/racci/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/racci/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/gzla1q2xfin7r8gnwpfdskhrihcig6l4-lix-2.93.3/share`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/pxnx7dbr0pz3qpgw2r315wdbb00afdw7-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
@@ -51,10 +51,10 @@ stdenv.mkDerivation {
       $out/bin/nvidia-persistenced
   '';
 
-  meta = with lib; {
-    homepage = "https://www.nvidia.com/object/unix.html";
-    description = "Settings application for NVIDIA graphics cards";
-    license = licenses.unfreeRedistributable;
+  meta = {
+    homepage = "https://github.com/NVIDIA/nvidia-persistenced";
+    description = "NVIDIA driver persistence daemon";
+    license = lib.licenses.mit;
     platforms = nvidia_x11.meta.platforms;
     maintainers = [ ];
     mainProgram = "nvidia-persistenced";

--- a/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
+++ b/pkgs/os-specific/linux/nvidia-x11/persistenced.nix
@@ -5,7 +5,9 @@ nvidia_x11: sha256:
   lib,
   fetchFromGitHub,
   m4,
+  glibc,
   libtirpc,
+  pkg-config,
 }:
 
 stdenv.mkDerivation {
@@ -19,8 +21,21 @@ stdenv.mkDerivation {
     inherit sha256;
   };
 
-  nativeBuildInputs = [ m4 ];
-  buildInputs = [ libtirpc ];
+  env = {
+    LIBRARY_PATH = "${glibc}/lib";
+    NIX_CFLAGS_COMPILE = toString [ "-I${libtirpc.dev}/include/tirpc" ];
+  };
+  NIX_LDFLAGS = [ "-ltirpc" ];
+
+  nativeBuildInputs = [
+    m4
+    pkg-config
+  ];
+
+  buildInputs = [
+    libtirpc
+    stdenv.cc.cc.lib
+  ];
 
   makeFlags = nvidia_x11.makeFlags ++ [ "DATE=true" ];
 
@@ -35,9 +50,6 @@ stdenv.mkDerivation {
     patchelf --set-rpath "$(patchelf --print-rpath $out/bin/nvidia-persistenced):${nvidia_x11}/lib" \
       $out/bin/nvidia-persistenced
   '';
-
-  env.NIX_CFLAGS_COMPILE = toString [ "-I${libtirpc.dev}/include/tirpc" ];
-  NIX_LDFLAGS = [ "-ltirpc" ];
 
   meta = with lib; {
     homepage = "https://www.nvidia.com/object/unix.html";

